src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,349 @@
           matchLength,
           matchLengthDifference: matchLength - inputLength,
           start: match.index,
           substring: match[0],
         });
-      }).filter(Boolean)
-    ),
+      }).filter(Boolean);
+    },
+    handleKeyEnter() {
+      if (
+        this.noResultsWereFound
+        || !this.userInput.length
+      ) return;
+      this.$router.push(this.filteredSymbols[this.focusedIndex].path);
+      this.closeQuickNavigationModal();
+    },
     orderSymbolsByPriority(matchingSymbols) {
       return matchingSymbols.sort((a, b) => {
         // Shortests symbol match title have preference over larger titles
         if (a.matchLengthDifference > b.matchLengthDifference) return 1;
         if (a.matchLengthDifference < b.matchLengthDifference) return -1;
-        // Shortests symbol title have preference over larger titles
-        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
-        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         // Matches at the beginning of string have relevance over matches at the end
         if (a.start > b.start) return 1;
         if (a.start < b.start) return -1;
+        // Shortests symbol title have preference over larger titles
+        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         return 0;
       });
     },
-  },
-  watch: {
-    userInput: 'debounceInput',
+    scrollIntoView() {
+      this.$refs.match[this.focusedIndex].scrollIntoView({
+        block: 'nearest',
+      });
+    },
+    startingPointHook() {
+      this.focusedIndex = this.totalItemsToNavigate - 1;
+    },
   },
 };
 </script>
 
 <style scoped lang="scss">
 @import 'docc-render/styles/_core.scss';
 
-  .close-icon {
-    padding-bottom: 10px;
-  }
-  .container {
-      flex-direction: row;
+$base-border-width: 1px;
+
+.quick-navigation {
+  input[type="text"] {
+    @include font-styles(body-large);
   }
-  .filter {
-    border-radius: $border-radius;
+  &__container {
     background-color: var(--color-fill);
-    border: solid var(--color-figure-gray) 1px;
-    padding: 20px;
-    width: 80%;
-    margin-bottom: 20px;
+    border: solid $base-border-width var(--color-fill-gray);
+    border-radius: $small-border-radius;
+    margin: 0 rem(16px);
+    > * {
+      --input-text: var(--color-figure-gray-secondary);
+    }
   }
-  .quick-navigation-modal {
-    background-color: var(--color-fill);
-    position: fixed;
-    top: 0;
-    bottom: 0;
-    left: 0;
-    right: 0;
-    z-index: 100;
-    padding: 100px;
+  &__filter{
+    --input-border-color: var(--color-fill);
+  }
+  &__magnifier-icon-container {
+    width: rem(18px);
+    > * {
+      width: 100%;
+    }
+    &.blue .magnifier-icon {
+      fill: var(--color-figure-blue);
+      color: var(--color-figure-blue);
+    }
+  }
+  &__match-list {
     overflow: scroll;
```

Comment:
In Chrome, with an macOS system config of "Show scroll bar... Automatically"

This overflow scroll is creating a jumping effect and also an extra padding in the bottom when there is no results.

https://user-images.githubusercontent.com/8567677/197261667-d3c58eef-3c4e-4393-ad88-818285b6f020.mov